UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/POSEndToEndFunctionalScenarios.cs

```diff
@@ -0,0 +1,95 @@
+﻿using NUnit.Framework;
+using UKHO.ExchangeSetService.API.FunctionalTests.Helpers;
+using UKHO.PeriodicOutputService.API.FunctionalTests.Helpers;
+using static UKHO.PeriodicOutputService.API.FunctionalTests.Helpers.TestConfiguration;
+
+namespace UKHO.PeriodicOutputService.API.FunctionalTests.FunctionalTests
+{
+    public class POSEndToEndFunctionalScenarios
+    {
+        private TestConfiguration config;
+        private string FssJwtToken;
+        public string userCredentialsBytes;
+        private POSWebJob WebJob;
+        private static readonly POSWebjobApiConfiguration POSWebJob = new TestConfiguration().POSWebJobConfig;
+
+        static FSSApiConfiguration FSSAuth = new TestConfiguration().FssConfig;
+        HttpResponseMessage POSWebJobApiResponse;
+        private List<string> DownloadedFolderPath;
+        public string IsoSha1BatchId = "f9523d33-ef12-4cc1-969d-8a95f094a48b";
+        public string ZipFilesBatchId = "483aa1b9-8a3b-49f2-bae9-759bb93b04d1";
+
+        [OneTimeSetUp]
+        public async Task Setup()
+        {
+            WebJob = new POSWebJob();
+            config = new TestConfiguration();
+            AuthTokenProvider authTokenProvider = new();
+            FssJwtToken = await authTokenProvider.GetFssToken();
+            string POSWebJobuserCredentialsBytes = CommonHelper.getbase64encodedcredentials(POSWebJob.userName, POSWebJob.password);
+            POSWebJobApiResponse = await WebJob.POSWebJobEndPoint(POSWebJob.baseUrl, POSWebJobuserCredentialsBytes);
+            await Task.Delay(120000); //As this functionality is realated to a webjob not an endpoint , so this is required to complete the webjob execution and then proceed further.
+        }
+
+        [Test]
+        public async Task WhenExtractedZipAndGeneratedISOAndSha1Files_ThenBatch1IsCreatedAndUploadedForISOAndSha1Files()
+        {
+            Assert.That((int)POSWebJobApiResponse.StatusCode, Is.EqualTo(202), $"Incorrect status code is returned {POSWebJobApiResponse.StatusCode}, instead of the expected status 202.");
+
+            HttpResponseMessage apiResponse = await GetBatchDetails.GetBatchDetailsEndpoint(FSSAuth.BaseUrl, IsoSha1BatchId);
+            Assert.That((int)apiResponse.StatusCode, Is.EqualTo(200), $"Incorrect status code is returned {apiResponse.StatusCode}, instead of the expected status 200.");
+
+            var BatchDetailsResponse = await CommonHelper.DeserializeAsyncResponse(apiResponse);
+
+            GetBatchDetails.GetBatchDetailsResponseValidation(BatchDetailsResponse);
+
+            GetBatchDetails.GetBatchDetailsResponseValidationForIsoAndSha1Files(BatchDetailsResponse);
```

Comment:
why are we putting two tests into one. your checking and asserting on a 202 then asserting on a 200 before validating on the response validation and then the iso validation

Reply:
Updated